packages/components/src/internal/components/chart/ChartFieldOption.tsx

```diff
@@ -38,70 +36,51 @@ export const getSelectOptions = (
         .map(col => ({ label: col.caption, value: col.fieldKey, data: col }));
 };
 
-export const shouldShowFieldOptions = (field: ChartFieldInfo, selectedType: ChartTypeInfo): boolean => {
-    const showForX = field.name === 'x' && (selectedType.name === 'scatter_plot' || selectedType.name === 'line_plot');
-    const showForY =
-        field.name === 'y' &&
-        (selectedType.name === 'scatter_plot' || selectedType.name === 'line_plot' || selectedType.name === 'box_plot');
-    return showForX || showForY;
-};
-
-const SCALE_TRANS_TYPES = [
-    { value: 'linear', label: 'Linear' },
-    { value: 'log', label: 'Log' },
-];
-
-const SCALE_RANGE_TYPES = [
-    { value: 'automatic', label: 'Automatic' },
-    { value: 'manual', label: 'Manual' },
-];
-
 const DEFAULT_SCALE_VALUES = { type: 'automatic', trans: 'linear' };
 
-interface ChartFieldOptionProps {
+interface OwnProps {
     field: ChartFieldInfo;
-    fieldValue?: SelectInputOption;
+    fieldValues?: Record<string, SelectInputOption>;
```

Comment:
The `fieldValues` is always passed, thus is not optional. If it may be undefined we should instead define it as:

`fieldValues: Record<string, SelectInputOption> | undefined`

The same is true for `scaleValues`.